### PR TITLE
Improvement/upgrade to angular 6

### PR DESCRIPTION
**Pull request by rayene**
Migrate to Angular 6 + updates all outdated dependencies to get rid of vulnerabilities + code adaptations.

### CODE REVIEW

**Comment by ovh-cds** [comment]
CDS Report build-image-preview#349.0 ✘  
* Build ui sources
  * Build Web UI ✘

File: webui/src/app/app.component.ts [hunk 1]
```diff
@@ -14,10 +14,11 @@ import { LoaderBean, LoadersStoreService } from './stores/loader-store.service';
 import { SidebarModule } from 'primeng/sidebar';
 import { ContentBean } from './models/commons/content-bean';
 import { ErrorBean, ErrorsStoreService, DropErrorAction, NewErrorAction } from './stores/errors-store.service';
-import { BehaviorSubject } from 'rxjs/BehaviorSubject';
+import { BehaviorSubject } from 'rxjs';
 import { ConfigStoreService, ConfigBean } from './stores/config-store.service';
 
 import { each, find } from 'lodash';
+import { Observable } from 'rxjs/internal/Observable';
```
Comment (FabienM) [inline webui/src/app/app.component.ts]: Sounds very weird to rely on someone else' internals! Auto-import typo for `rxjs/Observable`?

File: webui/src/app/components/appdetail/appdetail.component.ts [hunk 2]
```diff
@@ -7,10 +7,11 @@ import { Store } from '@ngrx/store';
 import { ApplicationsStoreService } from '../../stores/applications-store.service';
 import { EnvironmentsStoreService } from '../../stores/environments-store.service';
 import { element } from 'protractor';
-import { ISubscription } from 'rxjs/Subscription';
+import { SubscriptionLike as ISubscription } from 'rxjs';
 import { AutoUnsubscribe } from '../../shared/decorator/autoUnsubscribe';
 import { BadgeShieldsIOBean } from '../../widget/badgewidget/badgewidget.component';
 import {FieldsetModule} from 'primeng/fieldset';
+import { Observable } from 'rxjs/internal/Observable';
```
Comment (FabienM) [inline webui/src/app/components/appdetail/appdetail.component.ts]: Sounds very weird to rely on someone else' internals! Auto-import typo for `rxjs/Observable`?

File: webui/src/app/components/applications/applications.component.ts [hunk 3]
```diff
@@ -7,12 +7,13 @@ import { ContentListResponse, PageMetaData } from '../../models/commons/entity-b
...
 import { DataDeploymentService } from '../../services/data-deployment.service';
 import { UiKitPaginate } from '../../models/kit/paginate';
-import { BehaviorSubject } from 'rxjs/BehaviorSubject';
+import { BehaviorSubject } from 'rxjs';
 import { ApplicationsResolver } from '../../resolver/resolve-applications';
 import { ActivatedRoute, Router } from '@angular/router';
 import { TranslateService } from '@ngx-translate/core';
 import { OuiMessageComponent } from '../../kit/oui-message/oui-message.component';
 import { OuiPaginationComponent } from '../../kit/oui-pagination/oui-pagination.component';
+import { Observable } from 'rxjs/internal/Observable';
```
Comment (FabienM) [inline webui/src/app/components/applications/applications.component.ts]: Sounds very weird to rely on someone else' internals! Auto-import typo for `rxjs/Observable`?

File: webui/src/app/components/badges/badges.component.ts [hunk 4]
```diff
@@ -11,10 +11,11 @@ import { DataDeploymentService } from '../../services/data-deployment.service';
 import { UiKitPaginate } from '../../models/kit/paginate';
 import { ActivatedRoute } from '@angular/router';
 import { BadgesResolver } from '../../resolver/resolve-badges';
-import { BehaviorSubject } from 'rxjs/BehaviorSubject';
+import { BehaviorSubject } from 'rxjs';
 import { OuiPaginationComponent } from '../../kit/oui-pagination/oui-pagination.component';
 import { BadgeShieldsIOBean } from '../../widget/badgewidget/badgewidget.component';
 import { UIChart } from 'primeng/primeng';
+import { Observable } from 'rxjs/internal/Observable';
```
Comment (FabienM) [inline webui/src/app/components/badges/badges.component.ts]: Sounds very weird to rely on someone else' internals! Auto-import typo for `rxjs/Observable`?

File: webui/src/app/components/domains-browse/domains-browse.component.ts [hunk 5]
```diff
@@ -11,8 +11,9 @@ import { DataDeploymentService } from '../../services/data-deployment.service';
 import { UiKitPaginate } from '../../models/kit/paginate';
 import { ActivatedRoute } from '@angular/router';
 import { DomainsResolver } from '../../resolver/resolve-domains';
-import { BehaviorSubject } from 'rxjs/BehaviorSubject';
+import { BehaviorSubject } from 'rxjs';
 import { OuiPaginationComponent } from '../../kit/oui-pagination/oui-pagination.component';
+import { Observable } from 'rxjs/internal/Observable';
```
Comment (FabienM) [inline webui/src/app/components/domains-browse/domains-browse.component.ts]: Sounds very weird to rely on someone else' internals! Auto-import typo for `rxjs/Observable`?

File: webui/src/app/components/env-chip/env-chip.component.ts [hunk 6]
```diff
@@ -2,8 +2,9 @@ import { Component, Input, OnInit } from '@angular/core';
 import { EnvironmentsStoreService } from '../../stores/environments-store.service';
 import { EnvironmentBean } from '../../models/commons/applications-bean';
 import { Store } from '@ngrx/store';
-import { ISubscription } from 'rxjs/Subscription';
+import { SubscriptionLike as ISubscription } from 'rxjs';
 import { AutoUnsubscribe } from '../../shared/decorator/autoUnsubscribe';
+import { Observable } from 'rxjs/internal/Observable';
```
Comment (FabienM) [inline webui/src/app/components/env-chip/env-chip.component.ts]: Sounds very weird to rely on someone else' internals! Auto-import typo for `rxjs/Observable`?

File: webui/src/app/components/graphs/graph-browse.component.ts [hunk 7]
```diff
@@ -12,6 +12,7 @@ import { GraphBean, NodeBean, GraphVis } from '../../models/graph/graph-bean';
 import { DataDeploymentService } from '../../services/data-deployment.service';
 import { AutoUnsubscribe } from '../../shared/decorator/autoUnsubscribe';
 import { GraphComponent } from '../../widget/graph/graph.component';
+import { Observable } from 'rxjs/internal/Observable';
```
Comment (FabienM) [inline webui/src/app/components/graphs/graph-browse.component.ts]: Sounds very weird to rely on someone else' internals! Auto-import typo for `rxjs/Observable`?

File: webui/src/app/stores/applications-store.service.ts [hunk 8]
```diff
@@ -4,7 +4,8 @@ import { createFeatureSelector, createSelector, Selector, Store } from '@ngrx/st
 import { ActionWithPayloadAndPromise } from './action-with-payload';
 import { ApplicationBean, ApplicationPagesBean,
   DeploymentBean, DomainBean, DomainPagesBean, BadgeRatingBean } from '../models/commons/applications-bean';
-import { Subject } from 'rxjs/Subject';
+import { Subject } from 'rxjs';
+import { Observable } from 'rxjs/internal/Observable';
```
Comment (FabienM) [inline webui/src/app/stores/applications-store.service.ts]: Sounds very weird to rely on someone else' internals! Auto-import typo for `rxjs/Observable`?

File: webui/src/app/stores/badges-store.service.ts [hunk 9]
```diff
@@ -3,7 +3,8 @@ import { createFeatureSelector, createSelector, Selector, Store } from '@ngrx/st
 
 import { ActionWithPayloadAndPromise } from './action-with-payload';
 import {  BadgeBean, BadgePagesBean } from '../models/commons/badges-bean';
-import { Subject } from 'rxjs/Subject';
+import { Subject } from 'rxjs';
+import { Observable } from 'rxjs/internal/Observable';
```
Comment (FabienM) [inline webui/src/app/stores/badges-store.service.ts]: Sounds very weird to rely on someone else' internals! Auto-import typo for `rxjs/Observable`?

File: webui/src/app/stores/environments-store.service.ts [hunk 10]
```diff
@@ -3,6 +3,7 @@ import { createFeatureSelector, createSelector, MemoizedSelector, Store } from '
 
 import { ActionWithPayload } from './action-with-payload';
 import { EnvironmentBean } from '../models/commons/applications-bean';
+import { Observable } from 'rxjs/internal/Observable';
```
Comment (FabienM) [inline webui/src/app/stores/environments-store.service.ts]: Sounds very weird to rely on someone else' internals! Auto-import typo for `rxjs/Observable`?

File: webui/src/app/stores/graphs-store.service.ts [hunk 11]
```diff
@@ -3,8 +3,9 @@ import { createFeatureSelector, createSelector, Selector, Store } from '@ngrx/st
 
 import { ActionWithPayloadAndPromise } from './action-with-payload';
 import { ApplicationBean, ApplicationPagesBean, DeploymentBean, DomainBean, DomainPagesBean } from '../models/commons/applications-bean';
-import { Subject } from 'rxjs/Subject';
+import { Subject } from 'rxjs';
 import { GraphBean } from '../models/graph/graph-bean';
+import { Observable } from 'rxjs/internal/Observable';
```
Comment (FabienM) [inline webui/src/app/stores/graphs-store.service.ts]: Sounds very weird to rely on someone else' internals! Auto-import typo for `rxjs/Observable`?

File: webui/src/app/stores/help-store.service.ts [hunk 12]
```diff
@@ -1,13 +1,13 @@
...
 import { Injectable } from '@angular/core';
 import { createFeatureSelector, createSelector, Selector, Store } from '@ngrx/store';
 
 import { ActionWithPayloadAndPromise } from './action-with-payload';
-import { Subject } from 'rxjs/Subject';
 import { ContentBean } from '../models/commons/content-bean';
 import { DataContentService } from '../services/data-content.service';
 import { LoadersStoreService } from './loader-store.service';
 import { ErrorsStoreService, NewErrorAction, ErrorBean } from './errors-store.service';
+import { Observable } from 'rxjs/internal/Observable';
```
Comment (FabienM) [inline webui/src/app/stores/help-store.service.ts]: Sounds very weird to rely on someone else' internals! Auto-import typo for `rxjs/Observable`?

**Comment by FabienM** [comment]
From preview instance:

* direct access to a page won't work and redirect to the host root (`/applications/human-resources/travel-api`)
* assets URLs are broken (translations)

**Comment by sguiheux** [comment]
Observable must be import like this : import {Observable} from 'rxjs';   
cc @rayene @FabienM

**Comment by ovh-cds** [comment]
CDS Report build-image-preview#354.0 ✘  
* Build ui sources
  * Build Web UI ✘

**Comment by ovh-cds** [comment]
CDS Report build-image-preview#354.1 ✘  
* Build ui sources
  * Build Web UI ✘

**Comment by ovh-cds** [comment]
CDS Report build-image-preview#354.2 ✘  
* Build ui sources
  * Build Web UI ✘

**Comment by ovh-cds** [comment]
CDS Report build-image-preview#354.3 ✘  
* Build ui sources
  * Build Web UI ✘

**Pull request merged**